### PR TITLE
III-4465 - Location online

### PR DESCRIPTION
**Pull request by simon-debruijn**
### Added

- RadioButton type switch
- Online toggle and online url

### Changed

- Upgraded react-form-hook
- translations paths to be more consistent

---

Ticket: https://jira.uitdatabank.be/browse/III-4465

Screenshots:
Screenshot 2022-09-08 at 17 06 52

Screenshot 2022-09-08 at 17 06 41

### CODE REVIEW

File: src/i18n/fr.json [hunk 1]
```diff
@@ -88,7 +88,7 @@
     "step2": {
       "title": "Que voulez-vous ajouter"
     },
-    "municipality_and_place": {
+    "location": {
```
Comment (brampauwelyn) [inline src/i18n/fr.json:91]: Also add `online_url` and `is_online` translations here

File: src/i18n/de.json [hunk 2]
```diff
@@ -88,7 +88,7 @@
     "step2": {
       "title": "Was willst du hinzufügen"
     },
-    "municipality_and_place": {
+    "location": {
```
Comment (brampauwelyn) [inline src/i18n/de.json:91]: Also add `online_url` and `is_online` translations here

File: src/pages/create/EventForm.tsx [hunk 3]
```diff
@@ -2,14 +2,14 @@ import { useTranslation } from 'react-i18next';
 
 import { OfferType } from '@/constants/OfferType';
 import { SupportedLanguages } from '@/i18n/index';
-import { Event, isEvent } from '@/types/Event';
+import { AttendenceModes, Event, isEvent } from '@/types/Event';
```
Comment (brampauwelyn) [inline src/pages/create/EventForm.tsx:5]: Maybe it's better to use `AttendanceMode` singular? 
We also use `OfferType` and not `OfferTypes`?

```suggestion
import { AttendenceMode, Event, isEvent } from '@/types/Event';
```

File: src/pages/create/EventForm.tsx [hunk 4]
```diff
@@ -40,7 +42,8 @@ const EventForm = () => {
       event.location.address[i18n.language] ?? event.location.address;
     return {
       scope: isEvent(event) ? OfferType.EVENTS : OfferType.PLACES,
-      municipalityAndPlace: {
+      location: {
+        isOnline: event.attendanceMode === AttendenceModes.ONLINE,
```
Comment (brampauwelyn) [inline src/pages/create/EventForm.tsx:46]:
```suggestion
        isOnline: event.attendanceMode === AttendenceMode.ONLINE,
```

File: src/pages/steps/NameAndAgeRangeStep.tsx [hunk 5]
```diff
@@ -65,8 +65,11 @@ const NameAndAgeRangeStep = <TFormData extends FormDataUnion>({
 const nameAndAgeRangeStepConfiguration: StepsConfiguration<FormDataUnion> = {
   Component: NameAndAgeRangeStep,
   name: 'nameAndAgeRange',
-  title: (t) => t('create.step4.title'),
-  shouldShowStep: ({ watch }) => !!watch('municipalityAndPlace')?.place,
+  title: ({ t }) => t('create.name_and_age.title'),
+  shouldShowStep: ({ watch }) => {
+    const location = watch('location');
+    return !!location?.place || location?.isOnline;
```
Comment (brampauwelyn) [inline src/pages/steps/NameAndAgeRangeStep.tsx:71]:
```suggestion
    return !!location?.place || !!location?.isOnline;
```

**Review by brampauwelyn — APPROVED** [review]

File: src/pages/steps/AdditionalInformationStep.tsx [hunk 6]
```diff
@@ -200,7 +200,7 @@ AdditionalInformationStep.defaultProps = {
 
 const additionalInformationStepConfiguration: StepsConfiguration<FormDataUnion> = {
   Component: AdditionalInformationStep,
-  title: (t) => t(`movies.create.step5.title`),
+  title: ({ t }) => t(`movies.create.step5.title`),
```
Comment (brampauwelyn) [inline src/pages/steps/AdditionalInformationStep.tsx:203]: Change the title?
Now it's "Laat je film extra opvallen".
Maybe we could use the selected type in this title? Something like ` Laat je concert extra opvallen`

Comment (simon-debruijn) [reply src/pages/steps/AdditionalInformationStep.tsx:203]: I've sent it to @corneelwille, to maybe pickup in a followup ticket

Comment (corneelwille) [reply src/pages/steps/AdditionalInformationStep.tsx:203]: Good idea but we can indeed pick that up later. I already created a ticket: https://jira.uitdatabank.be/browse/III-4931. I think it works for `concert` but not for other eventtypes like `sport en beweging`: 
`Laat je sport en beweging extra opvallen` just sounds weird. But that's probably due to the copy of the eventtype, not the idea to make the title dynamic

**Pull request merged**